vyper/contracts/libs/FulfillmentApplier.vy

```diff
@@ -0,0 +1,471 @@
+                                   that no conduit should be used (and direct
+                                   approvals set on Consideration)
+
+    @return execution The transfer performed as a result of the fulfillment.
+    """
+    # Validate and aggregate consideration items on available orders and
+    # store result as a ReceivedItem.
+    receiveConsiderationItem: ReceivedItem = self._aggregateValidFulfillmentConsiderationItems(
+        ordersToExecute,
+        considerationComponents,
+        nextComponentIndex)
+
+
+    # Return execution for aggregated items provided by the fulfiller.
+    execution: Execution  = Execution({
+        item: receiveConsiderationItem,
+        offerer: msg.sender,
+        conduitKey: fulfillerConduitKey
+    })
+    return execution
+
+@internal
+@view
+def _aggregateAvailable(
+    ordersToExecute: DynArray[OrderToExecute, 10],
+    side: uint8,
+    fulfillmentComponents: DynArray[FulfillmentComponent, 10],
+    fulfillerConduitKey: bytes32
+) -> Execution:
+    """
+    @dev Internal view function to aggregate offer or consideration items
+         from a group of orders into a single execution via a supplied array
+         of fulfillment components. Items that are not available to aggregate
+         will not be included in the aggregated execution.
+
+    @param ordersToExecute       The orders to aggregate.
+    @param side                  The side (i.e. offer or consideration).
+    @param fulfillmentComponents An array designating item components to
+                                 aggregate if part of an available order.
+    @param fulfillerConduitKey   A bytes32 value indicating what conduit, if
+                                 any, to source the fulfiller's token
+                                 approvals from. The zero hash signifies that
+                                 no conduit should be used (and direct
+                                 approvals set on Consideration)
+
+    @return execution The transfer performed as a result of the fulfillment.
+    """
+    # Retrieve fulfillment components array length and place on stack.
+    totalFulfillmentComponents: uint256 = len(fulfillmentComponents)
+
+    # Ensure at least one fulfillment component has been supplied.
+    assert totalFulfillmentComponents != 0, "missing fulfillment component on aggregation"
+
+    # Determine component index after first available (0 implies none).
+    nextComponentIndex: uint256 = 0
+
+    # Iterate over components until finding one with a fulfilled order.
+    index: uint256 = 0
+    for component in fulfillmentComponents:
+        # Retrieve the fulfillment component index.
+        orderIndex: uint256 = component.orderIndex
+
+        # If order is being fulfilled (i.e. it is still available)...
+        if ordersToExecute[orderIndex].numerator != 0:
+            # Update the next potential component index.
+            nextComponentIndex = index + 1
```

Comment:
this thing with nextComponentIndex being `index + 1` and then removing the offset later on lines 459 and 469 is a bit odd. i would either initialize `nextComponentIndex` to something with some index that can never be hit (like MAX_UINT256), or have a separate `found: bool` flag to mark when this branch is hit.

Reply:
thanks for the suggestion. Updated it so that now `nextComponentIndex` is assigned just `index`, and used `index` to identify if the branch is hit.

Reply:
yes, just fixed it as well.